xtask/src/examples.rs

```diff
@@ -484,6 +484,55 @@ fn run_oak_functions_server(server: &Server) -> Box<dyn Runnable> {
     )
 }
 
+pub fn run_trusted_shuffler() -> Step {
+    let path_to_trusted_shuffler_scripts: PathBuf = [
+        env!("WORKSPACE_ROOT"),
+        "experimental",
+        "trusted_shuffler",
+        "scripts",
+    ]
+    .iter()
+    .collect();
+
+    // Run gRPC Trusted Shuffler.
+
+    // Build and run the echo gRPC backend.
+    let mut path_to_backend: PathBuf = path_to_trusted_shuffler_scripts.clone();
+    path_to_backend.push("run_grpc_backend");
+
+    let backend_cmd = Cmd::new("bash", &["-c", path_to_backend.to_str().unwrap()]);
+
+    // Build and run the Trusted Shuffler.
+    let mut path_to_trusted_shuffler: PathBuf = path_to_trusted_shuffler_scripts.clone();
+    path_to_trusted_shuffler.push("run_trusted_shuffler");
+
+    let trusted_shuffler_cmd =
+        Cmd::new("bash", &["-c", path_to_trusted_shuffler.to_str().unwrap()]);
+
+    // Build and run the echo gRPC client.
+    let mut path_to_client: PathBuf = path_to_trusted_shuffler_scripts;
+    path_to_client.push("run_single_grpc_client");
+
+    let client_cmd = Cmd::new("bash", &["-c", path_to_client.to_str().unwrap()]);
+
+    let client_step = Step::Single {
+        name: "Client".to_string(),
+        command: client_cmd,
+    };
+
+    let trusted_shuffler_step = Step::WithBackground {
+        name: "Trusted Shufflder and Backend".to_string(),
+        foreground: Box::new(client_step),
+        background: trusted_shuffler_cmd,
+    };
+
+    Step::WithBackground {
+        name: "Run echo gRPC Trusted Shuffler with k=1".to_string(),
+        foreground: Box::new(trusted_shuffler_step),
+        background: backend_cmd,
+    }
```

Comment:
@tiziano88 I believe I am using the `WithBackground` somehow wrongly---it seems xtask seems to get stuck here. Do you know how to use it?

Reply:
@tiziano88 xtask still gets stuck. Do you have an idea where I could start looking?

Reply:
Is this workig now, after removing the bash script layer, or does it still get stuck?

Reply:
Removing the indirection helped it not getting stuck, but sadly the backend server still returns with an error: https://github.com/project-oak/oak/runs/7910075586?check_suite_focus=true#step:6:139

I am not sure exactly why. This does not happen if I control+c the backend server locally. I wasn't sure where/how xtask ends the processes if [it is not interrupted](https://github.com/project-oak/oak/blob/a4d9157a9104d6f212f3f0cc23bdcc3e7dbfe09c/xtask/src/main.rs#L91).

Do you have a pointer?

Reply:
How did you confirm that it does not error out if you ctrl+c locally? AFAICT the backend server only handles ctrl+c correctly [in the HTTP case](https://github.com/project-oak/oak/blob/a4d9157a9104d6f212f3f0cc23bdcc3e7dbfe09c/experimental/trusted_shuffler/backend/src/main.rs#L84), not [in the gRPC](https://github.com/project-oak/oak/blob/a4d9157a9104d6f212f3f0cc23bdcc3e7dbfe09c/experimental/trusted_shuffler/backend/src/main.rs#L126-L129)

Reply:
My bad, thank you, I just tried it by locally ctrl+c'ing a server and it didn't seem unusal. It seems to work now!

Reply:
For the record, what you need to look at is the exit code of the binary, not any particular log message.